Sources/LogtoSocialPluginAlipay/LogtoSocialPluginAlipay.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  LogtoSocialPluginAlipay.swift
+//
+//
+//  Created by Gao Sun on 2022/3/31.
+//
+
+@_exported import AFServiceSDK
+import Foundation
+import LogtoSocialPlugin
+
+public enum LogtoSocialPluginAlipayError: LogtoSocialPluginError {
+    case authFailed(withCode: String)
+
+    public var code: String {
+        switch self {
+        case .authFailed:
+            return "auth_failed"
+        }
+    }
+
+    var localizedDescription: String {
+        switch self {
+        case let .authFailed(withCode):
+            return "Alipay auth failed with code \(withCode)."
+        }
+    }
+}
+
+// Follows Alipay official docs: https://opendocs.alipay.com/open/218/wy75xo
+public class LogtoSocialPluginAlipay: LogtoSocialPlugin {
+    public let urlSchemes: [String] = ["alipay"]
+
+    public init() {}
+
+    public func start(_ configuration: LogtoSocialPluginConfiguration) {
+        guard let redirectComponents = URLComponents(url: configuration.redirectUri, resolvingAgainstBaseURL: true)
+        else {
+            configuration.errorHandler(LogtoSocialPluginUriError.unableToConstructRedirectComponents)
+            return
+        }
+
+        guard var callbackComponents = URLComponents(url: configuration.callbackUri, resolvingAgainstBaseURL: true)
+        else {
+            configuration.errorHandler(LogtoSocialPluginUriError.unableToConstructCallbackComponents)
+            return
+        }
+
+        var components = URLComponents()
+
+        components.scheme = "https"
+        components.host = "authweb.alipay.com"
+        components.path = "/auth"
+        components.queryItems = [
+            URLQueryItem(name: "auth_type", value: "PURE_OAUTH_SDK"),
+            URLQueryItem(name: "scope", value: "auth_user"),
+        ] + (redirectComponents.queryItems ?? [])
+
+        AFServiceCenter.call(.auth, withParams: [
+            kAFServiceOptionBizParams: [
+                "url": components.string,
+            ],
+            kAFServiceOptionCallbackScheme: "logto-demo",
```

Comment:
not sure what is this for ,but wondering for production use, where should this come from?

Reply:
per PR description: to-do: [LOG-2084](https://linear.app/silverhand/issue/LOG-2084) to use global scheme for 3rd party app to go back